### DISCUSSION TITLE
Multiple package roots in Spack for non-root users

### DISCUSSION THREAD

**Discussion by giuseppe998e** [spack/spack#36126] [Q&A] [answered]
Hello,
I have access to a machine that uses spack as a package manager, the problem is that the `spack install ...` command is exclusive to the root account and therefore I cannot install the `rust` package.
In truth, I found that by changing the `config:install_tree:root` I can have packages installed to my home, but this way I no longer have access to the ones already installed, which forces me to reinstall those as well. Is there any way to make multiple roots co-exist?

**Comment by alalazo** [comment, marked as answer]
You don't need to use a `spack` command owned by sys-admin to use Spack. If you know where the system installations are, you can download your own Spack and set `upstreams` as explained [here](https://spack.readthedocs.io/en/latest/chain.html)[^1]

[^1]: To be fair that part of the docs needs some love!

**Reply by giuseppe998e** [reply]
I tried what you told me, but I used the same installation with the custom configurations in `$HOME`, it seems to work (at least `spack find` finds the packages), but is it still recommended to install your own spark?

**Reply by alalazo** [reply]
I would still use "my own" `spack` as a user, since I would have more control on it, while still benefiting from central installations with upstreams.
